src/Storages/ObjectStorage/StorageObjectStorage.h

```diff
@@ -203,6 +203,7 @@ class StorageObjectStorage::Configuration
     bool withGlobsIgnorePartitionWildcard() const;
     bool isPathWithGlobs() const;
     bool isNamespaceWithGlobs() const;
+    bool pathHasExactlyOneBracketsExpansion() const;
```

Comment:
IMHO, this class shouldn't contain this function. It is only about syntax, launching free function pathHasExactlyOneBracketsExpansion(conf.getPaths()) is more logical, IMO. WDYT?

Reply:
Yep, agree. Moreover, we don't even need this check, see thread above.